q2_feature_table/_filter.py

```diff
@@ -30,18 +29,25 @@ def f(data_vector, id_, metadata):
 
 
 def _filter(table, min_frequency, max_frequency, min_nonzero, max_nonzero,
-            metadata, where, axis):
+            metadata, where, axis, exclude_ids=False):
     if min_frequency == 0 and max_frequency is None and min_nonzero == 0 and\
        max_nonzero is None and metadata is None and where is None:
         raise ValueError("No filtering was requested.")
     if metadata is None and where is not None:
         raise ValueError("Metadata must be provided if 'where' is "
                          "specified.")
+    if metadata is None and exclude_ids is True:
+        raise ValueError("Metadata must be provided if 'exclude_ids' "
+                         "is specified")
 
     if metadata is not None:
         ids_to_keep = metadata.ids(where=where)
     else:
         ids_to_keep = table.ids(axis=axis)
+    if exclude_ids is True:
+        ids_to_keep = set(table.ids(axis=axis)) - set(ids_to_keep)
+    else:
+        ids_to_keep = set(ids_to_keep)
```

Comment:
If `exclude_ids` is True, we create a set.  If not, we need to turn `its_to_keep` into a set, since it might not be already.  This operation was previously in `_get_biom_filter_function`, but I think it makes sense here, since it highlights the fact that it's already a set in the case where we are doing the subtraction operation because `exclude_ids` is True.  If you disagree let me know; I'm happy to change it back.